include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -255,6 +255,57 @@ inline std::tuple<bool, int> compare_batch_cols(const batch_dim<2> &size1,
 }
 
 
+inline std::tuple<bool, int> check_batch_square(const batch_dim<> &size)
+{
+    if (size.stores_equal_sizes()) {
+        if (size.at(0)[0] != size.at(0)[1]) {
+            return std::tuple<bool, int>{false, 0};
+        } else {
+            return std::tuple<bool, int>{true, 0};
+        }
+    }
+
+    for (auto i = 0; i < size.get_num_batch_entries(); ++i) {
+        if (size.at(i)[1] != size.at(i)[1]) {
```

Comment:
I think this is supposed to be
```suggestion
        if (size.at(i)[1] != size.at(i)[0]) {
```